.github/workflows/dotnet.yml

```diff
@@ -3,7 +3,10 @@
 
 name: .NET build
 
-on: [push, pull_request]
+on:
+  push:
+  pull_request:
+    types: [opened, reopened]
```

Comment:
The default (i.e. if `types` isn't specified) is `[opened, reopened, synchronized]`.

The `synchronized` event happens whenever commits are pushed to a PR, which was redundant with us also triggering on all `push`es.

See also https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#pull_request

Reply:
> The `synchronized` event happens whenever commits are pushed to a PR, which was redundant with us also triggering on all `push`es.

IIRC they do that intentionally because the commit on the branch itself might be different from a merge commit when PR is out of date (not rebased), so the test result might be different as well? Admittedly, it's an edge case, but sometimes it might potentially catch issues early.

Reply:
Good point.. but it seems like this repo requires PRs to be up to date with master before merging, so I think it might not apply here?

Reply:
actually.. based on https://docs.github.com/en/webhooks/webhook-events-and-payloads?actionType=synchronize#pull_request, it looks like `synchronize` is probably only triggered when the head branch is pushed.

I think there might be a different option/event type for testing the merge commits? IIRC they're pushed to some other ref that gets CI tested.

Reply:
Oh huh. I probably confused the event types. But yeah, either way the base branch requirement takes care of that.